client/src/components/ProtectedRoute/AuthProtectedRoute.tsx

```diff
@@ -0,0 +1,16 @@
+import { useAuth } from '../../context/useAuthContext';
+import { Redirect, Route, RouteProps } from 'react-router-dom';
+
+interface AuthProtectedRouteProps extends RouteProps {
+  path: string;
+  exact: boolean;
+}
+
+const AuthProtectedRoute: React.FC<AuthProtectedRouteProps> = ({ ...rest }) => {
+  const { loggedInUser } = useAuth();
+
+  if (loggedInUser !== null && loggedInUser !== undefined) return <Redirect to="/dashboard" />;
```

Comment:
Just use `if (!loggedInUser)` for this since both `null` and `undefined` are all falsy :smile: